source/write-data-to-mongo.txt

```diff
@@ -0,0 +1,168 @@
+.. _java-rs-write:
+
+=====================
+Write Data to MongoDB
+=====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :description: Learn how to use the {+driver-short+} to write data to MongoDB.
+   :keywords: usage examples, save, crud, create, code example
+
+.. TODO: change toctree links when other pages are merged in.
+.. .. toctree::
+..    :titlesonly:
+..    :maxdepth: 1
+
+..    /write/insert
+..    /write/update
+..    /write/replace
+..    /write/delete
+..    /write/bulk-write
+..    /write/gridfs
+
+Overview
+--------
+
+This page contains copyable code examples of {+driver-short+} methods that you can use to
+write data to MongoDB.
+
+.. tip::
+
+  To learn more about any of the methods shown on this page, see the link
+  provided in each section.
+
+To use an example from this page, copy the code example into the
+:ref:`sample application <java-rs-write-sample>` or your own application.
+Be sure to replace all placeholders in the code examples, such as ``<connection string>``, with
+the relevant values for your MongoDB deployment.
+
+.. important:: Project Reactor Library
+
+   This guide uses the Project Reactor library to implement ``Publisher`` and
+   ``Subscriber`` methods. To learn more about the Project Reactor library and how
+   to use it, see `Getting Started
+   <https://projectreactor.io/docs/core/release/reference/#getting-started>`__
+   in the Reactor documentation.
+
+In {+java-rs+}, all operations that return a ``Publisher`` do not execute
+unless you subscribe to the returned ``Publisher`` method. This guide uses the
+``block()`` method to subscribe to all ``Publisher`` methods. To learn more
```

Comment:
One can subscribe to a `Publisher`, not to a method (regardless of whether it's a method on `Publisher` or on any other type).

```suggestion
unless you subscribe to the returned ``Publisher``. This guide uses the
``Mono.block()`` method to subscribe to a ``Publisher`` and block until it reaches its terminal state. To learn more
```

Reply:
> and block the ``Publisher`` until it reaches its terminal state

`block` does not block a `Publisher`, it blocks the current thread. You can either say what I proposed "and block until it reaches its terminal state", or "and block the current thread until the `Publisher` reaches its terminal state".

Reply:
- The Java drivers team also discussed and decided that a note like this should be moved into the `IMPORTANT` section that you already have on the page.
- Additionally, it may be worth mentioning the [cold `Publisher`](https://projectreactor.io/docs/core/release/reference/#reactor.hotCold) term there. We can say that all `Publishers` returned from the driver API are cold, and then explain that it means operations do not execute unless you subscribe bla-bla-bla.

Reply:
> all operations that return a Publisher are cold

It's the `Publisher`s that are cold, not operations. I'll suggest an edit for the whole `IMPORTANT` note as a new comment, as I think, it's also worth changing the wording in the first part of the note, where it talks about the Project Reactor.